server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -333,19 +337,47 @@ private MetaDataWithContent createMessageInOutbox(MessageWithId.CreationMessageE
     }
 
     private Flags getMessageFlags(CreationMessage message) {
+        Preconditions.checkState(!(hasKeywords(message) && isFlag(message)),
+                "Does not support both keywords and is*");
+        if (hasKeywords(message)) {
+            return buildFlagsFromKeywords(message.getKeywords().get());
+        } else {
+            return buildFlagsFromIsFlag(message);
+        }
+    }
+
+    private boolean isFlag(CreationMessage message) {
+        return message.isIsAnswered().isPresent()
+            || message.isIsDraft().isPresent()
+            || message.isIsFlagged().isPresent()
+            || message.isIsUnread().isPresent();
+    }
+
+    private boolean hasKeywords(CreationMessage message) {
+        return message.getKeywords().isPresent();
+    }
+
+    private Flags buildFlagsFromKeywords(ImmutableSet<JmapFlag> keywords) {
+        return keywords.stream()
+                .reduce(FlagsBuilder.builder(), JmapFlag::accumulator, JmapFlag::combiner)
+                .build();
+    }
+
+    private Flags buildFlagsFromIsFlag(CreationMessage message) {
         Flags result = new Flags();
-        if (!message.isIsUnread()) {
+        if (message.isIsUnread().isPresent() && !message.isIsUnread().get()) {
             result.add(Flags.Flag.SEEN);
         }
-        if (message.isIsFlagged()) {
+        if (message.isIsFlagged().isPresent() && message.isIsFlagged().get()) {
             result.add(Flags.Flag.FLAGGED);
         }
-        if (message.isIsAnswered() || message.getInReplyToMessageId().isPresent()) {
+        if ((message.isIsAnswered().isPresent() && message.isIsAnswered().get()) || message.getInReplyToMessageId().isPresent()) {
```

Comment:
If the message is in reply to, then this is not this message which is answered...

Also, alongside keywords JMAP spec removed the **inReplyTo** concept.

It became: 

```
 If the message has an In-Reply-To header, the server SHOULD add the $Answered keyword to all messages with the corresponding Message-Id header at this point.
```